### ISSUE TITLE
feature request: Add an option to open Oil only when you open a directory, not a file

### ISSUE THREAD

**Issue by ABEC-projects** [stevearc/oil.nvim#368]
### Did you check existing requests?

- [X] I have searched the existing issues

### Describe the feature

I want an option not to open Oil window every time you try to edit file with 'nvim path/to/file' and instead have the default nvim behavior for that (just opening that file)

### Provide background

It's very annoying that every time I want to open a file I get Oil instead of file buffer. E. g. I want to edit my .zshrc, I type `nvim ~/.zshrc` and now I have to hit `<leader>g.` to show hidden files and only after that I can open it. Another reason I doesn't like this behavior is that I cannot open a new file with just `nvim ~/new.txt` because Oil won't find it. I've tried finding the solution, but got nothing. I really like oil.nvim, so I'd like to have this feature

### What is the significance of this feature?

strongly desired

**Comment by stevearc** [comment]
You are telling me that `nvim <file>` opens oil instead of editing the file? That is absolutely not the correct behavior. Please open another issue as a bug report. The bug report issue template has a form for collecting proper information for debugging a problem.

**Comment by ABEC-projects** [comment]
This was not a bug. I just accidentally added `require("oil").open()` so it was opening Oil every time I entered nvim.
